src/libraries/Microsoft.Extensions.Logging.Console/src/TextWriterExtensions.cs

```diff
@@ -0,0 +1,132 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal static class TextWriterExtensions
+    {
+        public static void WriteColoredMessage(this TextWriter textWriter, string message, ConsoleColors colors, bool disableColors)
+        {
+            bool colorChanged = textWriter.SetBackgroundColor(colors.Background, disableColors);
+            colorChanged = textWriter.SetForegroundColor(colors.Foreground, disableColors) || colorChanged;
+            textWriter.Write(message);
+            if (colorChanged)
+            {
+                textWriter.ResetColor();
+            }
+        }
+
+        private static bool SetBackgroundColor(this TextWriter textWriter, ConsoleColor? background, bool disableColors = false)
+        {
+            if (disableColors)
+            {
+                return false;
+            }
+            if (background.HasValue)
+            {
+                textWriter.Write(GetBackgroundColorEscapeCode(background.Value));
+            }
+            else
+            {
+                textWriter.Write(DefaultBackgroundColor);
+            }
+            return true;
+        }
+
+        private static bool SetForegroundColor(this TextWriter textWriter, ConsoleColor? foreground, bool disableColors = false)
+        {
+            if (disableColors)
+            {
+                return false;
+            }
+            if (foreground.HasValue)
+            {
+                textWriter.Write(GetForegroundColorEscapeCode(foreground.Value));
+            }
+            else
+            {
+                textWriter.Write(DefaultForegroundColor);
+            }
+            return true;
+        }
+
+        private static void ResetColor(this TextWriter textWriter)
+        {
+            textWriter.SetBackgroundColor(null);
+            textWriter.SetForegroundColor(null);
+        }
+
+        private const string DefaultForegroundColor = "\x1B[39m\x1B[22m"; // reset to default foreground color
+        private const string DefaultBackgroundColor = "\x1B[49m"; // reset to the background color
+
+        private static string GetForegroundColorEscapeCode(ConsoleColor color)
```

Comment:
>GetForegroundColorEscapeCode [](start = 30, length = 28)

I think this method and background method can use simple arrays instead of switch statement.